Hey @ashb @mik-laj @kaxil @turbaszek - > I need your help. I have this really strange problem and maybe you will be able to help quickly. This is the LAST step to have stable selective tests working. 

The MySQL tests in this build fail because of fernet.InvalidToken and I can't figure it out quickly:

It's rather easy to reproduce:

```
./breeze --backend sqlite --db-reset 
```

And then: 
```
pytest tests/providers/amazon/aws/hooks/test_emr.py::TestEmrHook::test_create_job_flow_extra_args
```

This fails with: 

```
____________________________________________________________________________________________________________________________ TestEmrHook.test_create_job_flow_extra_args _____________________________________________________________________________________________________________________________

self = <tests.providers.amazon.aws.hooks.test_emr.TestEmrHook testMethod=test_create_job_flow_extra_args>

    @mock_emr
    def test_create_job_flow_extra_args(self):
        """
        Test that we can add extra arguments to the launch call.
    
        This is useful for when AWS add new options, such as
        "SecurityConfiguration" so that we don't have to change our code
        """
        client = boto3.client('emr', region_name='us-east-1')
    
        hook = EmrHook(aws_conn_id='aws_default', emr_conn_id='emr_default')
        # AmiVersion is really old and almost no one will use it anymore, but
        # it's one of the "optional" request params that moto supports - it's
        # coverage of EMR isn't 100% it turns out.
>       cluster = hook.create_job_flow({'Name': 'test_cluster', 'ReleaseLabel': '', 'AmiVersion': '3.2'})

tests/providers/amazon/aws/hooks/test_emr.py:62: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
airflow/providers/amazon/aws/hooks/emr.py:80: in create_job_flow
    config = emr_conn.extra_dejson.copy()
airflow/models/connection.py:377: in extra_dejson
    if self.extra:
/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/attributes.py:358: in __get__
    retval = self.descriptor.__get__(instance, owner)
airflow/models/connection.py:292: in get_extra
    return fernet.decrypt(bytes(self._extra, 'utf-8')).decode()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cryptography.fernet.MultiFernet object at 0x7fa7aa0ebed0>
msg = b'gAAAAABfjpjdT5OM4KF8yhsX_-nEC3-53b-XDTF4hpZVTzqVrYjEh5AHebgky5PCvt_hogmCInRNWXm5Zhe_zK-x9-TgAYxQlKZ2UwBcTaTi4uufY0ji...Ppw6xt39aGU0rQF_NbZYSXL0Ls9XceeALWIuGjEuxtpEGmlyX1bV2vAS9dU36rzffPcZj0VM8zvpbn3gNaFeZBq0aSSKPHtYOtzsWFj8qVdxQsellPHyI=', ttl = None

    def decrypt(self, msg, ttl=None):
        for f in self._fernets:
            try:
                return f.decrypt(msg, ttl)
            except InvalidToken:
                pass
>       raise InvalidToken
E       cryptography.fernet.InvalidToken

/usr/local/lib/python3.7/site-packages/cryptography/fernet.py:183: InvalidToken
--------------------------------------------------------------------------------------------------------------------------------------- Captured stdout setup ----------------------------------------------------------------------------------------------------------------------------------------
========================= AIRFLOW ==========================
Home of the user: /root
Airflow home /root/airflow
Skipping initializing of the DB as it was initialized already.
You can re-initialize the database by adding --with-db-init flag when running tests.
---------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------------
[2020-10-20 07:59:31,472] {credentials.py:1087} INFO - Found credentials in environment variables.
----------------------------------------------------------------------------------------------------------------------------------------- Captured log call ------------------------------------------------------------------------------------------------------------------------------------------
INFO     botocore.credentials:credentials.py:1087 Found credentials in environment variables.
====================================================================================================================================== short test summary info =======================================================================================================================================
FAILED tests/providers/amazon/aws/hooks/test_emr.py::TestEmrHook::test_create_job_flow_extra_args - cryptography.fernet.InvalidToken

```

Does it ring a bell maybe ??????

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -442,6 +442,7 @@ jobs:
       POSTGRES_VERSION: ${{ matrix.postgres-version }}
       RUN_TESTS: true
       TEST_TYPES: ${{needs.build-info.outputs.testTypes}}
+      TEST_TYPE: ""
     if: >
         needs.build-info.outputs.run-tests == 'true' &&
         (github.repository == 'apache/airflow' || github.event_name != 'schedule')
@@ -494,6 +495,7 @@ jobs:
       MYSQL_VERSION: ${{ matrix.mysql-version }}
       RUN_TESTS: true
       TEST_TYPES: ${{needs.build-info.outputs.testTypes}}
+      TEST_TYPE: ""
     if: >
         needs.build-info.outputs.run-tests == 'true' &&
         (github.repository == 'apache/airflow' || github.event_name != 'schedule')
@@ -543,6 +545,7 @@ jobs:
       PYTHON_MAJOR_MINOR_VERSION: ${{ matrix.python-version }}
       RUN_TESTS: true
       TEST_TYPES: ${{needs.build-info.outputs.testTypes}}
+      TEST_TYPE: ""
     if: >
         needs.build-info.outputs.run-tests == 'true' &&
         (github.repository == 'apache/airflow' || github.event_name != 'schedule')
@@ -596,6 +599,7 @@ jobs:
       POSTGRES_VERSION: ${{needs.build-info.outputs.defaultPostgresVersion}}
       RUN_TESTS: true
       TEST_TYPES: "Quarantined"
+      TEST_TYPE: ""
       NUM_RUNS: 10
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     if: >

--- a/breeze
+++ b/breeze
@@ -173,6 +173,9 @@ function breeze::setup_default_breeze_constants() {
     export SCREEN_WIDTH
     readonly SCREEN_WIDTH
 
+    # for Breeze default tests executed are "All"
+    export TEST_TYPE=${TEST_TYPE:="All"}
+
     # Update short and long options in the breeze-complete script
     # This way autocomplete will work automatically with all options available
     # shellcheck source=breeze-complete

--- a/scripts/ci/libraries/_initialization.sh
+++ b/scripts/ci/libraries/_initialization.sh
@@ -461,7 +461,7 @@ function initialization::initialize_github_variables() {
 }
 
 function initialization::initialize_test_variables() {
-    export TEST_TYPE=${TEST_TYPE:="All"}
+    export TEST_TYPE=${TEST_TYPE:=""}
 }
 
 # Common environment that is initialized by both Breeze and CI scripts

--- a/scripts/ci/testing/ci_run_airflow_testing.sh
+++ b/scripts/ci/testing/ci_run_airflow_testing.sh
@@ -37,7 +37,7 @@ function run_airflow_testing_in_docker() {
         echo "Making sure docker-compose is down"
         echo
         docker-compose --log-level INFO -f "${SCRIPTS_CI_DIR}/docker-compose/base.yml" \
-            down --remove-orphans --timeout 10
+            down --remove-orphans --volumes --timeout 10
         echo
         echo "System-prune docker"
         echo
@@ -53,6 +53,12 @@ function run_airflow_testing_in_docker() {
         echo
         echo "Starting try number ${try_num}"
         echo
+        if [[ " ${ENABLED_INTEGRATIONS} " =~ " kerberos " ]]; then
+            echo "Creating Kerberos network"
+            kerberos::create_kerberos_network
+        else
+            echo "Skip creating kerberos network"
+        fi
         docker-compose --log-level INFO \
           -f "${SCRIPTS_CI_DIR}/docker-compose/base.yml" \
           -f "${SCRIPTS_CI_DIR}/docker-compose/backend-${BACKEND}.yml" \
@@ -60,6 +66,10 @@ function run_airflow_testing_in_docker() {
           "${DOCKER_COMPOSE_LOCAL[@]}" \
              run airflow "${@}"
         exit_code=$?
+        if [[ " ${INTEGRATIONS[*]} " =~ " kerberos " ]]; then
+            echo "Delete kerberos network"
+            kerberos::delete_kerberos_network
+        fi
         if [[ ${exit_code} == 254 ]]; then
             echo
             echo "Failed starting integration on ${try_num} try. Wiping-out docker-compose remnants"
@@ -149,13 +159,6 @@ if [[ ${TEST_TYPE=} != "" ]]; then
 fi
 readonly TEST_TYPES
 
-export RUN_INTEGRATION_TESTS=${RUN_INTEGRATION_TESTS:=""}
-export ENABLED_INTEGRATIONS=${ENABLED_INTEGRATIONS:=""}
-
-if [[ " ${ENABLED_INTEGRATIONS[*]} " =~ " kerberos " ]]; then
-    kerberos::create_kerberos_network
-fi
-
 echo "Running TEST_TYPES: ${TEST_TYPES}"
 
 for TEST_TYPE in ${TEST_TYPES}
@@ -166,6 +169,9 @@ do
     if [[ ${TEST_TYPE:=} == "Integration" ]]; then
         export ENABLED_INTEGRATIONS="${AVAILABLE_INTEGRATIONS}"
         export RUN_INTEGRATION_TESTS="${AVAILABLE_INTEGRATIONS}"
+    else
+        export ENABLED_INTEGRATIONS=""
+        export RUN_INTEGRATION_TESTS=""
     fi
 
     for _INT in ${ENABLED_INTEGRATIONS}
@@ -174,11 +180,11 @@ do
         INTEGRATIONS+=("${SCRIPTS_CI_DIR}/docker-compose/integration-${_INT}.yml")
     done
 
-
     export TEST_TYPE
+
     echo "**********************************************************************************************"
     echo
-    echo "                            TEST_TYPE: ${TEST_TYPE}"
+    echo "      TEST_TYPE: ${TEST_TYPE}, ENABLED INTEGRATIONS: ${ENABLED_INTEGRATIONS}"
     echo
     echo "**********************************************************************************************"
 

--- a/scripts/in_container/entrypoint_ci.sh
+++ b/scripts/in_container/entrypoint_ci.sh
@@ -185,6 +185,7 @@ if [[ "${GITHUB_ACTIONS}" == "true" ]]; then
         # timeouts in seconds for individual tests
         "--setup-timeout=20"
         "--execution-timeout=60"
+        "--with-db-init"
         "--teardown-timeout=20"
         # Only display summary for non-expected case
         # f - failed

--- a/tests/bats/breeze/test_breeze_complete.bats
+++ b/tests/bats/breeze/test_breeze_complete.bats
@@ -255,10 +255,10 @@
   assert_equal "${_breeze_default_postgres_version}" "${POSTGRES_VERSION}"
 }
 
-@test "Test default test type same as TEST_TYPE" {
+@test "Test default test type is empty" {
   load ../bats_utils
   #shellcheck source=breeze-complete
   source "${AIRFLOW_SOURCES}/breeze-complete"
 
-  assert_equal "${_breeze_default_test_type}" "${TEST_TYPE}"
+  assert_equal "" "${TEST_TYPE}"
 }

